WordPress/gutenberg
# Experiment: Load the template when editing a page

## Description

An experiment for editing the full page template and content when loading a page in the editor.

My main intention with this PR is to explore how we can resolve #29821, so that special pages like the "Posts page" are represented more accurately in the editor. As a side effect, this allows editing the full template of any page.

This PR has some hacks that would need to be resolved before a change of this sort could be merged.

## How has this been tested?

- Set a Posts page in Settings > Reading if you don't already have one set
- Note that you'll need content to exist in that page already, otherwise you can't load the block editor (see https://core.trac.wordpress.org/ticket/45537)
- Try editing the Post page, or any other page

## Screenshots 

**Editing the Posts page**

https://user-images.githubusercontent.com/1699996/119416107-8bc6d480-bcb8-11eb-842f-1d6286adeabc.mov

**Loading a "normal" page in the editor**

https://user-images.githubusercontent.com/1699996/119416223-d0eb0680-bcb8-11eb-954e-4d302f103d08.mov

## Types of changes




Experimental

## Checklist:
